### PR TITLE
pulpcore-plugin 0.1.0rc4 compatibility 

### PR DESCRIPTION
**Pull request by dkliban**
This PR adds compatibility with pulpcore-plugin 0.1.0rc4, introduces black formatting, and introduces towncrier for changes.

### CODE REVIEW

File: .travis/mariadb.yml [hunk 1]
```diff
@@ -1,2 +1,9 @@
+# WARNING: DO NOT EDIT!
+#
+# This file was generated by plugin_template, and is managed by bootstrap.py. Please use
+# bootstrap.py to update this file.
+#
+# For more info visit https://github.com/pulp/plugin_template
+
```
Comment (goosemania) [inline .travis/mariadb.yml]: Is this file still needed? mariadb.yml

File: .travis/publish_plugin_pypi.sh [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+# This file was generated by plugin_template, and is managed by bootstrap.py. Please use
+# bootstrap.py to update this file.
+#
+# For more info visit https://github.com/pulp/plugin_template
+
 pip install twine
 
 python setup.py sdist bdist_wheel --python-tag py3
-twine upload dist/* -u pulp -p $PYPI_PASSWORD
+twine upload dist/* -u None -p $PYPI_PASSWORD
```
Comment (goosemania) [inline .travis/publish_plugin_pypi.sh]: Why is username changed to None?

File: docs/workflows/index.rst [hunk 3]
```diff
@@ -28,6 +28,4 @@ set is the hostname and port::
 .. toctree::
    :maxdepth: 2
 
-   sync
-   upload
-   publish-host
+   cache
```
Comment (goosemania) [inline docs/workflows/index.rst]: why did you keep only cache use case?

File: template_config.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+coverage: False
+deploy_client_to_pypi: True
+deploy_client_to_rubygems: True
+deploy_daily_client_to_pypi: True
+deploy_daily_client_to_rubygems: True
+deploy_to_pypi: True
+docs_test: True
+pydocstyle: True
+test_bindings: False
+pypi_username: None
```
Comment (goosemania) [inline template_config.yml]: Is it correct that the username is None?

File: CHANGES/5217.bugfix [hunk 5]
```diff
@@ -0,0 +1 @@
+Updates serializers and tests to make plugin compatible with pulpcore-plugin 0.1.0rc4.
```
Comment (goosemania) [inline CHANGES/5217.bugfix]: How about a .feature about adding towncrier? Just a suggestion.

Comment (goosemania) [inline CHANGES/5217.bugfix]: I'm not sure what was incompatible with pulpcore, except for tests.
I didn't notice any changes in serializer (maybe black changes make it hard to notice).

File: template_config.yml (hunk 4 above)
Comment (dkliban) [reply template_config.yml]: No, I need to correct this.

File: docs/workflows/index.rst (hunk 3 above)
Comment (dkliban) [reply docs/workflows/index.rst]: Because there is only one workflow right now. It's the cache one.
